bridge/src/connectors/connector_c.rs

```
@@ -227,108 +113,215 @@ impl ConnectorC {
-        };
+        }
+    }
 
-        spend_info_cache
+    fn lock_script_cache(&self, leaf_index: usize) -> Option<LockScriptCache> {
+        match lock_script_cache_id(&self.commitment_public_keys, leaf_index) {
+            Ok(cache_id) => Some(
+                TAPROOT_LOCK_SCRIPTS_CACHE
+                    .write()
+                    .unwrap()
+                    .get_or_put(cache_id, || {
+                        let (script, control_block) = generate_script_and_control_block(
+                            self.operator_taproot_public_key,
+                            &self.lock_scripts_bytes(),
+                            leaf_index,
+                        );
+                        let encoded_data = bitcode::encode(script.as_bytes());
+                        let compressed_data = compress(&encoded_data, DEFAULT_COMPRESSION_LEVEL)
+                            .expect("Unable to compress script for caching");
+                        LockScriptCache {
+                            control_block,
+                            encoded_script: compressed_data,
+                        }
+                    })
+                    .clone(),
+            ),
+            _ => None,
+        }
     }
 
-    pub fn cache_id(
-        commitment_public_keys: &BTreeMap<CommitmentMessageId, WinternitzPublicKey>,
-    ) -> Result<String, ConnectorError> {
-        let first_winternitz_public_key = commitment_public_keys.iter().next();
+    fn lock_scripts_bytes(&self) -> Vec<Vec<u8>> {
+        match spend_info_cache_id(&self.commitment_public_keys) {
+            Ok(cache_id) => {
+                let file_path = get_lock_scripts_cache_path(&cache_id);
+                let lock_scripts_bytes = read_disk_cache(&file_path)
+                    .inspect_err(|e| {
+                        eprintln!(
+                            "Failed to read lock scripts cache from expected location: {}",
+                            e
+                        );
```

If this is `ErrorKind::NotFound`, it's an expected scenario - either we're generating lock scripts for the first time, or the user has cleared the cache, requiring regeneration. In this case, it's safe to ignore the error and just continue.

---

Not only not found, it also can be a decoding or decompressing error, which means the data on disk is either corrupted or no longer be compatible with current code.

---

My point was that on that specific error, we can and should ignore it. On any other error (that you describe) do what we're doing, which is output to error output.

---

oh ok, Fixed